### PR TITLE
integrate the messagebus and send access elements through messagebus to kafka.

### PR DESCRIPTION
**Pull request by hopkings2008**

### CODE REVIEW

**Comment by hopkings2008** [comment]
@thesues @cxt90730 Please help to review this PR, this PR is to send the access message to kafka through message bus and i have tested by using s3cmd.

File: api/access-log-handler.go [hunk 1]
```diff
@@ -49,6 +52,45 @@ func (a AccessLogHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	response := newReplacer.Replace(a.format)
 
 	helper.AccessLogger.Println(5, response)
+	// send the entrys in access logger to message bus.
+	elems := newReplacer.Get()
+	a.notify(elems)
+}
+
+func (a AccessLogHandler) notify(elems map[string]string) {
+	if 0 == helper.CONFIG.MsgBus.Enabled {
```
[the remark this answers is not in the document]
Comment (hopkings2008) [reply api/access-log-handler.go]: ok, just learned a new syntax for go. :)

File: helper/config.go [hunk 2]
```diff
@@ -83,8 +83,13 @@ type KMSConfig struct {
 }
 
 type MsgBusConfig struct {
-	Type   int                    `toml:"msg_bus_type"`
-	Server map[string]interface{} `toml:"msg_bus_server"`
+	Enabled          int                    `toml:"msg_bus_enable"`
```
[the remark this answers is not in the document]
Comment (hopkings2008) [reply helper/config.go]: ok, bool is meanful. I will make a ament.

File: api/log-replacer.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ import (
 type Replacer interface {
 	Replace(string) string
 	Set(key, value string)
+	Get() map[string]string
```
[the remark this answers is not in the document]
Comment (hopkings2008) [reply api/log-replacer.go]: ok, i will use the name GetReplaceValues. thanks for you suggestion.

[the remark this answers is not in the document]
Comment (hopkings2008) [reply api/log-replacer.go]: @cannium, please help to review this PR again, i have made some enhancements according to your suggestions. thx.

**Comment by thesues** [comment]
@hopkings2008 canyou rebase this patch to resolve conflicts?

**Comment by hopkings2008** [comment]
> @hopkings2008 canyou rebase this patch to resolve conflicts?

@thesues Please check it again. i have sync this PR with master branch.

**Pull request merged**